mmhuman3d/data/datasets/pipelines/transforms.py

```diff
@@ -678,8 +748,35 @@ def __call__(self, results):
         r = results['rotation']
         trans = get_affine_transform(c, s, r, self.image_size)
 
+
+
         if 'img' in results:
             img = results['img']
+
+            # img before affine
+            ori_img = img.copy()
+            (h, w) = ori_img.shape[:2]
+            (cX, cY) = (w // 2, h // 2)
+            M = cv2.getRotationMatrix2D((cX, cY), r, 1.0)
+            cos = np.abs(M[0, 0])
+            sin = np.abs(M[0, 1])
+            # compute the new bounding dimensions of the image
+            nW = int((h * sin) + (w * cos))
+            nH = int((h * cos) + (w * sin))
+            # adjust the rotation matrix to take into account translation
+            M[0, 2] += (nW / 2) - cX
+            M[1, 2] += (nH / 2) - cY
+            # perform the actual rotation and return the image
+            ori_img = cv2.warpAffine(ori_img, M, (nW, nH))
+            M = np.concatenate([M,np.array([[0.0,0.0,1.0]])])
+            M = np.linalg.inv(M)
+            N = np.concatenate([trans,np.array([[0.0,0.0,1.0]])])
+            crop_trans = np.dot(N,M)
+            results['crop_transform'] = crop_trans[:2]
+
```

Comment:
This part does not seem to fit here well. Why do we need to add these lines if `img` and `ori_img` are already available (line 780-784)?

Reply:
Transform function `Rotation` added.